### PR TITLE
product feature added

### PR DESCRIPTION
**Pull request by Rahul16-creator**
Product Feature added

### CODE REVIEW

File: src/main/java/com/shopping_app/shoppingApp/model/Product/ProductResponse.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    private double productPrice;
+
+    private String productDescription;
+
+    private int availableQuantity;
+
+    public static ProductResponse from(Product product) {
+        if (product == null) {
+            return null;
+        }
```
Comment (drashtivpatel) [inline src/main/java/com/shopping_app/shoppingApp/model/Product/ProductResponse.java]: Do we need this?

File: src/main/java/com/shopping_app/shoppingApp/service/ProductService.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        List<Product> products = productRepository.findAll();
+        return products.stream()
+                .map(ProductResponse::from)
+                .collect(Collectors.toList());
+    }
+
+    public ProductResponse getProductById(Long id) {
+        Optional<Product> product = productRepository.findById(id);
+        if (product.isEmpty()) {
+            throw new NotFoundException("Product Not Found in our database Please check product Id", HttpStatus.NOT_FOUND);
```
Comment (drashtivpatel) [inline src/main/java/com/shopping_app/shoppingApp/service/ProductService.java]:
```suggestion
            throw new NotFoundException("Product not found", HttpStatus.NOT_FOUND);
```

File: src/main/java/com/shopping_app/shoppingApp/controller/ProductController.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    public ApiResponse getResponse(String message, Object data) {
+        return ApiResponse.builder()
+                .data(data)
+                .code(HttpStatus.OK)
+                .message(message)
+                .status(ResponseType.SUCCESS)
+                .build();
+    }
```
Comment (drashtivpatel) [inline src/main/java/com/shopping_app/shoppingApp/controller/ProductController.java]: This block of code is repeated, can you extract it to a utils file?

File: src/main/java/com/shopping_app/shoppingApp/model/Product/ProductResponse.java (hunk 1 above)
Comment (Rahul16-creator) [reply src/main/java/com/shopping_app/shoppingApp/model/Product/ProductResponse.java]: Yes this is not neccessary , while i copying the code from already implemented branch i missed out this to remove .

File: src/main/java/com/shopping_app/shoppingApp/controller/ProductController.java (hunk 3 above)
Comment (Rahul16-creator) [reply src/main/java/com/shopping_app/shoppingApp/controller/ProductController.java]: yeah , Sure

File: src/main/java/com/shopping_app/shoppingApp/model/Product/ProductResponse.java (hunk 1 above)
Comment (Rahul16-creator) [reply src/main/java/com/shopping_app/shoppingApp/model/Product/ProductResponse.java]: updated

**Review by drashtivpatel — APPROVED** [review]

**Pull request merged**